model_analyzer/state/analyzer_state_manager.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.state.analyzer_state import AnalyzerState
+from model_analyzer.model_analyzer_exceptions \
+    import TritonModelAnalyzerException
+
+import signal
+import logging
+import pickle
+import os
+import glob
+
+
+class AnalyzerStateManager:
+    """
+    Maintains the state of the Model Analyzer
+    """
+    def __init__(self, config):
```

Comment:
Can you just pass the config variables required instead of the whole config object?

Reply:
Sure but is there a particular reason for this?

Reply:
I think it makes the object's dependencies clear but not a very big deal.